src/server/pkg/backoff/retry.go

```diff
@@ -102,12 +102,16 @@ func RetryUntilCancel(ctx context.Context, operation Operation, b BackOff, notif
 
 	b.Reset()
 	for {
-		if err = operation(); err == nil {
+		err = operation()
+		if err == nil {
 			return nil
 		}
 		if ctx.Err() != nil {
 			return ctx.Err() // return if cancel() was called inside operation()
 		}
+		if errors.Is(err, ErrContinue) {
+			b.Reset()
+		}
```

Comment:
This is definitely closer to how I expected `ErrContinue` to behave. I think it's worth updating its comment to make it clear that it no longer requires `NotifyContinue`. I see a couple of places in our codebase where `b.Reset()` is explicitly called in the loop to get this behavior, but I'm assuming no one would use the `notify` function to decide whether to turn this effect on or off.

Reply:
Done